mssql_python/cursor.py

```
@@ -104,6 +104,49 @@ def __init__(self, connection, timeout: int = 0) -> None:
 
         self.messages = []  # Store diagnostic messages
 
+    def _get_encoding_settings(self):
+        """
+        Get the encoding settings from the connection.
+        
+        Returns:
+            dict: A dictionary with 'encoding' and 'ctype' keys, or default settings if not available
+        """
+        if hasattr(self._connection, 'getencoding'):
+            try:
+                return self._connection.getencoding()
+            except:
+                # Return default encoding settings if there's an error
+                return {
+                    'encoding': 'utf-16le', 
+                    'ctype': ddbc_sql_const.SQL_WCHAR.value
+                }
+        # Return default encoding settings if getencoding is not available
+        return {
+            'encoding': 'utf-16le',
+            'ctype': ddbc_sql_const.SQL_WCHAR.value
+        }
+
+    def _get_decoding_settings(self, sql_type):
+        """
+        Get decoding settings for a specific SQL type.
+        
+        Args:
+            sql_type: SQL type constant (SQL_CHAR, SQL_WCHAR, etc.)
+            
+        Returns:
+            Dictionary containing the decoding settings.
+        """
+        try:
+            # Get decoding settings from connection for this SQL type
+            return self._connection.getdecoding(sql_type)
+        except Exception as e:
```

This too... This except is too broad. If getencoding() or getdecoding() fails due to a programming bug (AttributeError, logic error, etc.), we silently force a fallback encoding (utf-16le), which can mask real defects.